site/app/libraries/FileUtils.php

```diff
@@ -502,32 +514,62 @@ public static function validateUploadedFiles($files) {
         for ($i = 0; $i < $num_files; $i++) {
             //extract the values from each file
             $name = $files['name'][$i];
-            $type = mime_content_type($files['tmp_name'][$i]);
-            $size = $files['size'][$i];
-            $err_msg = "";
+            $tmp_name = $files['tmp_name'][$i];
+            $type = mime_content_type($tmp_name);
+            
+            $zip_status = FileUtils::getZipFileStatus($tmp_name);
+            $errors = [ErrorMessages::uploadErrors($files['error'][$i])];
+
+            //check if its a zip file
+            $is_zip = $type === 'application/zip';
+            if ($is_zip) {
+                $zip_status = FileUtils::getZipFileStatus($tmp_name);
+                if ($zip_status !== \ZipArchive::ER_OK) {
+                    $errors[] = ErrorMessages::getZipErrorMessage($zip_status);
+                }
+                else {
+                    $size = FileUtils::getZipSize($tmp_name);
+                    if (!FileUtils::checkFileInZipName($tmp_name)) {
+                        $errors[] = "Invalid filename within zip file";
+                    }
+                }
+            }
 
-            //did anything go wrong?
-            $err_msg = ErrorMessages::uploadErrors($files['error'][$i]);
+            //for zip files use the size of the contents in case it gets extracted
+            $size = $is_zip ? FileUtils::getZipSize($tmp_name) : $files['size'][$i];
 
             //manually check against set size limit
             //incase the max POST size is greater than max file size
             if ($size > $max_size) {
-                $err_msg = "File \"" . $name . "\" too large got (" . Utils::formatBytes("mb", $size) . ")";
+                $errors[] = "File \"" . $name . "\" too large got (" . Utils::formatBytes("mb", $size) . ")";
             }
 
             //check filename
             if (!FileUtils::isValidFileName($name)) {
-                $err_msg = "Invalid filename";
+                $errors[] = "Invalid filename";
+            }
+
+            $errors = array_unique($errors);
+            $success = true;
+            foreach ($errors as $err) {
+                if ($err !== 'No error.') {
+                    $success = false;
+                }
+            }
```

Comment:
Success should be that `$success = count($errors) > 0`. Having `'No error.'` be an entry just begs to be confusing to someone.